src/pages/EditRequestTagPage.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Navigation from '../libs/Navigation/Navigation';
+import useLocalize from '../hooks/useLocalize';
+import ScreenWrapper from '../components/ScreenWrapper';
+import HeaderWithBackButton from '../components/HeaderWithBackButton';
+import TagPicker from '../components/TagPicker';
+
+const propTypes = {
+    /** Transaction default tag value */
+    defaultTag: PropTypes.string.isRequired,
+
+    /** The policyID we are getting tags for */
+    policyID: PropTypes.string.isRequired,
+
+    /** The tag name to which the default tag belongs to */
+    tagName: PropTypes.string.isRequired,
+
+    /** Callback to fire when the Save button is pressed  */
+    onSubmit: PropTypes.func.isRequired,
+};
+
+function EditRequestTagPage({defaultTag, policyID, tagName, onSubmit}) {
+    const {translate} = useLocalize();
+
+    const selectTag = (tag) => {
+        onSubmit({tag: tag.searchText});
+    };
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            shouldEnableMaxHeight
+        >
+            <HeaderWithBackButton
+                title={tagName || translate('common.tag')}
```

Comment:
Hi, we should have set the tag name here too - 

https://github.com/Expensify/App/blob/27e2efbfd7ba64e5ee81f11789f0615b540d7ec3/src/libs/ReportUtils.js#L1776

it would have prevented issue https://github.com/Expensify/App/issues/30377#issuecomment-1780011566